purchases/src/main/kotlin/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -919,13 +914,21 @@ internal class BillingWrapper(
         }
     }
 
-    private fun BillingClient.queryPurchasesAsyncWithTracking(
+    private fun BillingClient.queryPurchasesAsyncWithTrackingEnsuringOneResponse(
         @BillingClient.ProductType productType: String,
         queryParams: QueryPurchasesParams,
         listener: PurchasesResponseListener,
     ) {
+        val hasResponded = AtomicBoolean(false)
```

Comment:
Oh right, so this is the main change right? We weren't handling multiple results from the billing client on the `queryPurchases` method?

Reply:
Yes exactly. I haven't been able to reproduce it but forcing the callback in this function to execute twice will render the same exception. I looked at the other places where we could be triggering the callback twice and it's only in the backend call, but didn't see anything strange there.